crates/holochain/src/core/workflow/validation_receipt_workflow.rs

```diff
@@ -19,37 +18,41 @@ use holochain_zome_types::block::CellBlockReason;
 #[cfg(test)]
 mod tests;
 
+#[cfg(test)]
+mod unit_tests;
+
 pub async fn pending_receipts(
     vault: &DbRead<DbKindDht>,
     validators: Vec<AgentPubKey>,
 ) -> StateQueryResult<Vec<(ValidationReceipt, AgentPubKey, DhtOpHash)>> {
     vault
-        .read_async({ move |txn| get_pending_validation_receipts(&txn, validators) })
+        .read_async(move |txn| get_pending_validation_receipts(&txn, validators))
         .await
 }
 
-#[instrument(skip(vault, network, keystore, conductor))]
+#[instrument(skip(vault, network, keystore, apply_block))]
 /// Send validation receipts to their authors in serial and without waiting for
 /// responses.
 /// TODO: Currently still waiting for responses because we don't have a network call
 /// that doesn't.
-pub async fn validation_receipt_workflow(
+pub async fn validation_receipt_workflow<B>(
     dna_hash: Arc<DnaHash>,
     vault: DbWrite<DbKindDht>,
-    network: HolochainP2pDna,
+    network: impl HolochainP2pDnaT,
     keystore: MetaLairClient,
-    conductor: ConductorHandle,
-) -> WorkflowResult<WorkComplete> {
-    // Who we are.
-    // TODO: I think this is right but maybe we need to make sure these cells are in
-    // running apps?.
-    let cell_ids = conductor.running_cell_ids(Some(CellStatus::Joined));
-
-    if cell_ids.is_empty() {
+    running_cell_ids: HashSet<CellId>,
+    apply_block: B,
```

Comment:
Removed dependency on the conductor handle because it's hard to mock and we don't actually need a lot of functionality from it